src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriter.java

```diff
@@ -19,16 +18,17 @@
 public class FileWriter implements Closeable {
 
     private static final Logger log = LoggerFactory.getLogger(KustoSinkTask.class);
-    FileDescriptor currentFile;
+    FileProperties currentFile;
     private Timer timer;
-    private Consumer<FileDescriptor> onRollCallback;
+    private Consumer<FileProperties> onRollCallback;
     private final long flushInterval;
     private final boolean shouldCompressData;
     private Supplier<String> getFilePath;
     private OutputStream outputStream;
     private String basePath;
     private CountingOutputStream countingStream;
     private long fileThreshold;
+    private FileDescriptor currentFd;
```

Comment:
```suggestion
    private FileDescriptor currentFileDescriptor;
```

no need to abbrivate